osu.Game/Overlays/Mods/ModSelectOverlay.cs

```diff
@@ -450,18 +450,30 @@ private void filterMods()
                 modState.ValidForSelection.Value = modState.Mod.Type != ModType.System && modState.Mod.HasImplementation && IsValidMod.Invoke(modState.Mod);
         }
 
-        private void updateRankingInformation()
+        /// <summary>
+        /// Updates any information displayed on the overlay regarding the effects of the selected mods.
+        /// </summary>
+        /// <param name="mods">The list of mods to show effect from. This can be overriden to include effect of mods that are not part of the <see cref="SelectedMods"/> bindable (e.g. room mods in multiplayer/playlists).</param>
+        protected virtual void UpdateOverlayInformation(IReadOnlyList<Mod> mods)
         {
-            if (rankingInformationDisplay == null)
-                return;
+            if (rankingInformationDisplay != null)
+            {
+                double multiplier = 1.0;
 
-            double multiplier = 1.0;
+                foreach (var mod in mods)
+                    multiplier *= mod.ScoreMultiplier;
 
-            foreach (var mod in SelectedMods.Value)
-                multiplier *= mod.ScoreMultiplier;
+                rankingInformationDisplay.ModMultiplier.Value = multiplier;
+                rankingInformationDisplay.Ranked.Value = mods.All(m => m.Ranked);
+            }
 
-            rankingInformationDisplay.ModMultiplier.Value = multiplier;
-            rankingInformationDisplay.Ranked.Value = SelectedMods.Value.All(m => m.Ranked);
+            if (beatmapAttributesDisplay != null)
+            {
+                if (!ReferenceEquals(beatmapAttributesDisplay.Mods.Value, mods))
+                    beatmapAttributesDisplay.Mods.Value = mods;
+                else
+                    beatmapAttributesDisplay.Mods.TriggerChange(); // mods list may be same but a mod setting has changed, trigger change in that case.
```

Comment:
I don't like this having to be handled out-of-band like that. Was there an issue with letting `BeatmapAttributesDisplay` handle this internally? It already had a `ModSettingChangeTracker` in place.

Or, if that _is_ an issue, then basically delete bindable usage from `BeatmapAttributesDisplay`  and have it recalculate everything every time it is told to. Something like `UpdateDisplay(IEnumerable<Mod>)`.

Reply:
There is no issue with the old method, just thought I would save another usage of `ModSettingChangeTracker` with this. I agree it's bad though now that I look at it again, I'll revert to the previous logic.